### DISCUSSION TITLE
Open PDF file from memorystream or buffer

### DISCUSSION THREAD

**Discussion by Mohamad777** [sumatrapdfreader/sumatrapdf#2371] [Q&A]
In some cases developers try not to save a file on disk.
I know that it's possible in muPDF to open a file from memory (buffer). I'm using it in my Android application.
Is it possible in sumatraPDF to open a PDF file from memory? And How?

I didn't find any full documentation for developers.

Thanks in advacnce.

**Comment by GitHubRulesOK** [comment]
Its not an api and I am not dev but my understanding is SumatraPDF in common with any PDF reader like MuPDF will work with any acceptable FILE format if it is a named FS (file stream) object but the file must be 100% long with a **fixed number of bytes** in the file allocation table just like on a disk, Thus any file reader can acess the decimal adressed chunks, whether those be PNG chunks or PDF streamed objects. 
The downstream exception that I know of is PDFTron where after the first file request the streams can be random requested chunks of served linearised pdf that are used and discarded in pieces much like an audio/video stream, but that still requires a file of byte length to be read to be served from a servers File System.
In short most file formats must be a FILE name of given length to read, even if an arbitary temporary cache name.

**Comment by kjk** [comment]
That's not really possible on Windows. It might make sense on Android but Windows doesn't have an easy way to send memory from one process to another.

If you want to open a file in Sumatra from your program, save it to disk and open that file.
